modules/auth/oauth2/oauth2.go

```diff
@@ -2,22 +2,30 @@ package oauth2
 
 import (
 	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/log"
 	"github.com/gorilla/sessions"
 	"github.com/markbates/goth"
 	"github.com/markbates/goth/gothic"
 	"github.com/markbates/goth/providers/github"
 	"net/http"
 	"os"
 	"github.com/satori/go.uuid"
+	"path/filepath"
 )
 
 var (
-	sessionUsersStoreKey = "gitea-oauth-sessions"
-	providerHeaderKey = "gitea-oauth-provider"
+	sessionUsersStoreKey = "gitea-oauth2-sessions"
+	providerHeaderKey    = "gitea-oauth2-provider"
 )
 
 func init() {
-	gothic.Store = sessions.NewFilesystemStore(os.TempDir(), []byte(sessionUsersStoreKey))
+	dir, _ := setting.WorkDir()
+	tmpDir := filepath.Join(dir, "data", "sessions", "oauth2")
```

Comment:
And since it's not really temporary (you never delete it) it should be called `sessionDir` 🙂